packages/web3-eth-contract/src/utils.ts

```diff
@@ -227,6 +227,29 @@ export const getCreateAccessListParams = ({
 	return txParams;
 };
 
+/**
+ * Generates the Ethereum address of a contract created via a regular transaction.
+ *
+ * This function calculates the contract address based on the sender's address and nonce,
+ * following Ethereum's address generation rules.
+ *
+ * @param from The sender’s Ethereum {@link Address}, from which the contract will be deployed.
+ * @param nonce The transaction count (or {@link Numbers}) of the sender account at the time of contract creation.
+ *              This is used to ensure uniqueness of the generated address.
+ * @returns An Ethereum {@link Address} of the contract in checksum address format.
+ * @throws An {@link InvalidAddressError} if the provided address ('from') is invalid.
+ * @throws An {@link InvalidNumberError} if the provided nonce value is not in a valid format.
+ * @example
+ * ```ts
+ * const from = "0x1234567890abcdef1234567890abcdef12345678";
+ * const nonce = 1; // The nonce value for the transaction
```

Comment:
This is probably more helpful, but please ensure it works and is correct before accepting the suggestion.

```suggestion
 * const nonce = (await web3.eth.getTransactionCount(from)) + 1; // The nonce value for the transaction
```

Reply:
Although nonce is a basic knowledge that all developers must have, it is better to understand